src/data/resolvers/colonyActions.ts

```diff
@@ -0,0 +1,180 @@
+import { Context } from '~context/index';
+import { ColonyActions, ColonyAndExtensionsEvents } from '~types/index';
+
+export const colonyActionsResolvers = ({
+  colonyManager: { networkClient },
+  colonyManager,
+}: Required<Context>): Resolvers => ({
+  Query: {
+    async colonyAction(_, { transactionHash, colonyAddress }) {
+      const { provider } = networkClient;
+
+      /*
+       * Get all clients from all extensions enabled in the cololony
+       */
+      const clientsInstancesArray = (
+        await Promise.all(
+          Object.values(ClientType).map(async (clientType) => {
+            try {
+              return await colonyManager.getClient(clientType, colonyAddress);
+            } catch (error) {
+              return undefined;
+            }
+          }),
+        )
+      ).filter((clientType) => !!clientType);
+
+      /*
+       * Get the colony client specifically
+       */
+      const colonyClient = clientsInstancesArray.find(
+        (client) => client?.clientType === ClientType.ColonyClient,
+      );
+
+      /*
+       * Try to get the transaction receipt. If the transaction is mined, you'll
+       * get a return from this call, otherwise, it's null.
+       */
+      const transactionReceipt = await provider.getTransactionReceipt(
+        transactionHash,
+      );
+
+      if (transactionReceipt) {
+        const {
+          transactionHash: hash,
+          status,
+          logs,
+          blockHash,
+          from,
+        } = transactionReceipt;
+
+        /*
+         * Get the block time in ms
+         *
+         * we fallback to 0, which is 1/1/1970 :)
+         * If we don't find a time for the current tx (which shouldn't happen actually)
+         */
+        const createdAt = blockHash
+          ? await getBlockTime(provider, blockHash)
+          : 0;
+
+        /*
+         * @NOTE Parse all logs with all clients to generate all the possible events
+         * This is the second iteration of this implementation.
+         *
+         * It is less perfomant than just iterating over the clientsInstancesArray and
+         * then just passing it the logs array to process, but this way allows us to
+         * preseve the position of each event as it was originally emmited by the contracts.
+         *
+         * This way we can use that index position to inferr the action type from them
+         */
+        const reverseSortedEvents = logs
+          ?.map((log) => {
+            const [parsedLog] = clientsInstancesArray
+              .map((clientType) => {
+                const type = clientType?.clientType;
+                const potentialParsedLog = clientType?.interface.parseLog(log);
+                if (potentialParsedLog) {
+                  const { name, values } = potentialParsedLog;
+                  return {
+                    name,
+                    values,
+                    createdAt,
+                    emmitedBy: type,
```

Comment:
so if type is null then we pass here null, and from what i saw above emitedBy is required, so we probably should add to if potentiatlParsedLog && type

Reply:
Type cannot be `null`. There is always a `clientType` on the client instance